suite-native/module-authorize-device/src/screens/connect/UnlockedDeviceRequired.tsx

```diff
@@ -0,0 +1,79 @@
+import { useCallback } from 'react';
+import { useSelector } from 'react-redux';
+
+import { useFocusEffect } from '@react-navigation/native';
+
+import {
+    selectIsBluetoothSupportedByDevice,
+    selectIsDeviceAuthorized,
+    selectIsDeviceConnected,
+    selectIsDeviceConnectedViaBluetooth,
+    selectIsDevicePinLocked,
+} from '@suite-common/wallet-core';
+import { selectBluetoothPermissionStatus } from '@suite-native/bluetooth';
+import {
+    ConnectAndUnlockDeviceScreenContent,
+    TurnOnAndUnlockDeviceScreenContent,
+} from '@suite-native/device';
+import { FeatureFlag, useFeatureFlag } from '@suite-native/feature-flags';
+import {
+    AuthorizeDeviceStackParamList,
+    AuthorizeDeviceStackRoutes,
+    RootStackParamList,
+    Screen,
+    StackToStackCompositeScreenProps,
+} from '@suite-native/navigation';
+
+import { ConnectDeviceScreenHeader } from '../../components/connect/ConnectDeviceScreenHeader';
+
+export const UnlockedDeviceRequiredScreen = ({
+    navigation,
+    route: { params },
+}: StackToStackCompositeScreenProps<
+    AuthorizeDeviceStackParamList,
+    AuthorizeDeviceStackRoutes.UnlockedDeviceRequired,
+    RootStackParamList
+>) => {
+    const isDeviceConnected = useSelector(selectIsDeviceConnected);
+    const isDevicePinLocked = useSelector(selectIsDevicePinLocked);
+    const isDeviceAuthorized = useSelector(selectIsDeviceAuthorized);
+
+    const isBluetoothEnabled = useFeatureFlag(FeatureFlag.IsBluetoothEnabled);
+    const bluetoothPermissionStatus = useSelector(selectBluetoothPermissionStatus);
+    const isDeviceConnectedViaBluetooth = useSelector(selectIsDeviceConnectedViaBluetooth);
+    const isBluetoothSupportedByDevice = useSelector(selectIsBluetoothSupportedByDevice);
+
+    const isBluetoothVariantVisible =
+        isBluetoothEnabled &&
+        bluetoothPermissionStatus === 'granted' &&
```

Comment:
Nit: this could be a selector, not a big issue though.

Reply:
It's the only place where it's used and only relevant to this specific screen. Not sure how I'd call such a selector.

Reply:
`isBluetoothPermissionGranted` would be my recommendation, but as I said, it's really just a nit. Feel free to ignore. Absolutely not critical now.